src/mapper.rs

```diff
@@ -52,3 +52,13 @@ pub trait Mapper {
     /// ```
     fn unmap(&mut self, virt_start: usize, bytes: usize);
 }
+
+/// an identity mapper which maps a physical address into itself.
+#[derive(Clone, Copy, Debug)]
```

Comment:
As per https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits, could you change the code to derive all possible traits?